invertigo.asl

```diff
@@ -1,106 +1,75 @@
+    vars.CoffeeBreakName = "2_coffee_break";
+    vars.MenuName = "menu_v4";
+
     Assembly.Load(File.ReadAllBytes("Components/asl-help")).CreateInstance("Unity");
+    vars.Helper.GameName = "Invertigo";
     vars.Helper.LoadSceneManager = true;
 
     settings.Add("resetMenu", true, "Automatically reset when in main menu.");
     settings.Add("resetCoffeeBreak", true, "Automatically reset when restarting while in Coffee Break.");
 
-    if (timer.CurrentTimingMethod == TimingMethod.RealTime) {
-        if (MessageBox.Show(
-            "Invertigo uses the time spent in the level as the timing method.\n"+
-            "LiveSplit is currently set to RTA, do you want to switch?",
-            "Invertigo AutoSplitter",
-            MessageBoxButtons.YesNo,
-            MessageBoxIcon.Question
-        ) == DialogResult.Yes) {
-            timer.CurrentTimingMethod = TimingMethod.GameTime;
-        }
-    }
+    vars.Helper.AlertGameTime();
 }
 
 init
 {
-    vars.CoffeeBreakName = "2_coffee_break";
-    vars.MenuName = "menu_v4";
     vars.Helper.TryLoad = (Func<dynamic, bool>)(mono =>
-        {
-            //Thanks to CaptainRektbeard for figuring out the params...
-            vars.Helper["Timer"] = mono.Make<float>("ScoreManager", 1, "_instance", "_timer");
-            vars.Helper["IsTimerRunning"] = mono.Make<bool>("ScoreManager", 1, "_instance", "_running");
-            vars.Helper["IsGamePaused"] = mono.Make<bool>("PauseManager", 1, "_instance", "Paused");
-            return true;
-        });
-    //Init some vars
-    current.PrevLevels = 0.0;
-    current.PrevTimes = 0.0;
-    current.Timer = 0.0;
-    current.Scene = null;
-}
+    {
+        vars.Helper["Timer"] = mono.Make<float>("ScoreManager", 1, "_instance", "_timer");
+        vars.Helper["IsTimerRunning"] = mono.Make<bool>("ScoreManager", 1, "_instance", "_running");
 
-gameTime
-{
-    return TimeSpan.FromSeconds(current.Timer + current.PrevTimes + current.PrevLevels);
+        return true;
+    });
+
+    vars.TotalTime = 0; 
 }
 
 update
 {
-    current.Scene = vars.Helper.Scenes.Active.Name;
-    if (current.Scene == null) {
-        current.Scene = old.Scene;
-    }
-    current.DidReset = old.Timer != null && old.Timer > current.Timer;
-    current.DidLevelChange = current.Scene != old.Scene;
-    if (current.DidReset)
-    {
-        if (current.DidLevelChange)
-        {
-            current.PrevLevels += current.PrevTimes + old.Timer;
-            current.PrevTimes = 0.0;
-        }
-        else
-        {
-            current.PrevTimes += old.Timer;
-        }
-    }
+    current.Scene = vars.Helper.Scenes.Active.Name ?? old.Scene;
 }
 
 start
 {
-    if (current.Scene == vars.CoffeeBreakName && current.IsTimerRunning)
-    {
-        current.PrevLevels = 0.0;
-        current.PrevTimes = 0.0;
-        return true;
-    }
+    return current.Scene == vars.CoffeeBreakName && !old.IsTimerRunning && current.IsTimerRunning;
```

Comment:
Only return `true` or `false` here. Handle other start-related logic in `onStart`.